### PR TITLE
[blog] Symfony 3.3 DI init

### PR DESCRIPTION
**Pull request by TomasVotruba**

### CODE REVIEW

**Comment by TomasVotruba** [comment]
@javiereguiluz Hey Javier, just wondering, if you got 3 minutes, could you check this?

I want to make it as easy as possible for people to ABSORB :thought_balloon:  new Symfony 3.3 DI knowledge.

File: source/_posts/2017/2017-05-05-how-to-refactor-to-new-dependency-injection-features-in-symfony-3-3.md [hunk 1]
```diff
@@ -0,0 +1,251 @@
+---
+layout: post
+title: "How to refactor to new Dependency Injection features in Symfony 3.3"
+perex: '''
+    This May will be released Symfony 3.3 with many DependencyInjection improvements.
+    Each of them is quite nice, <strong>but combined together - they are huge jump</strong> compare to what we have now.
+    <br><br>
+    Today I will you what code can you drop and how to migrate it.
```
Comment (martinahynkova) [inline source/_posts/2017/2017-05-05-how-to-refactor-to-new-dependency-injection-features-in-symfony-3-3.md]: I will "show" you

File: source/_posts/2017/2017-05-05-how-to-refactor-to-new-dependency-injection-features-in-symfony-3-3.md [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+    - Using Nette? Check tools [by F3l1x](https://github.com/contributte/di) and [Pavel Janda](https://github.com/ublaboo/directory-register)
+- [action method injection](https://github.com/symfony/symfony/pull/21771) - known [from Laravel](https://mattstauffer.co/blog/laravel-5.0-method-injection#solution)
+- [abstract controller](https://github.com/symfony/symfony/pull/22157)
+
+**Those bold** will be important part of every Symfony 3.3+ application. You can click on them to get to post or PR, where they're explained in
+more detailed way. But I think there is **quicker way to learn them**...
+
+<br>
+
+### "A full code examples is worth ten thousand words of explanation."
```
Comment (martinahynkova) [inline source/_posts/2017/2017-05-05-how-to-refactor-to-new-dependency-injection-features-in-symfony-3-3.md]: example

File: source/_posts/2017/2017-05-05-how-to-refactor-to-new-dependency-injection-features-in-symfony-3-3.md [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+            calls:
+                - ["setEntityManager", ["@entity_manager"]]
+        
+    App\:
+        resource: ../{ Controller, Command, Subscriber, Repository }
+
+    App\SomeService: ~
+```
+
+That awesome, isn't it?
```
Comment (martinahynkova) [inline source/_posts/2017/2017-05-05-how-to-refactor-to-new-dependency-injection-features-in-symfony-3-3.md]: That is awesome

File: source/_posts/2017/2017-05-05-how-to-refactor-to-new-dependency-injection-features-in-symfony-3-3.md [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+
+Now made full use of Symfony 3.3 Dependency Injection features.
```
Comment (martinahynkova) [inline source/_posts/2017/2017-05-05-how-to-refactor-to-new-dependency-injection-features-in-symfony-3-3.md]: Now made full ......? co to je za větu, nerozumím....

File: source/_posts/2017/2017-05-05-how-to-refactor-to-new-dependency-injection-features-in-symfony-3-3.md [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+
+```yaml
+# app/config/services.yml
+services:
+    _defaults:
+        autowired: true
+        autoconfigure: true
+        
+    App\: # no more manual registration of similar groups of services 
+        resource: ../{ Controller, Command, Subscriber }
```
Comment (javiereguiluz) [inline source/_posts/2017/2017-05-05-how-to-refactor-to-new-dependency-injection-features-in-symfony-3-3.md]: I'm not sure white spaces are allowed here. I always see it like this:

```
resource: ../{Controller,Command,Subscriber}
```

File: source/_posts/2017/2017-05-05-how-to-refactor-to-new-dependency-injection-features-in-symfony-3-3.md [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+        autowired: true
+        autoconfigure: true
+
+    _instanceof: # clean and explicit dependency injection to abstract services
+        App\Repository\AbstractRepository:
+            calls:
+                - ["setEntityManager", ["@entity_manager"]]
+        
+    App\:
+        resource: ../{ Controller, Command, Subscriber, Repository }
```
Comment (javiereguiluz) [inline source/_posts/2017/2017-05-05-how-to-refactor-to-new-dependency-injection-features-in-symfony-3-3.md]: Same comment here about white spaces

File: source/_posts/2017/2017-05-05-how-to-refactor-to-new-dependency-injection-features-in-symfony-3-3.md (hunk 4 above)
Comment (javiereguiluz) [inline source/_posts/2017/2017-05-05-how-to-refactor-to-new-dependency-injection-features-in-symfony-3-3.md]: Now you are using all the shiny new Symfony 3.3 Dependency Injection features.

File: source/_posts/2017/2017-05-05-how-to-refactor-to-new-dependency-injection-features-in-symfony-3-3.md [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+
+    App\SomeService: ~
+```
+
+That awesome, isn't it?
+
+Now made full use of Symfony 3.3 Dependency Injection features.
+
+**Do you like it? Do you plan to skip some of those and stick to the current version?**
```
Comment (javiereguiluz) [inline source/_posts/2017/2017-05-05-how-to-refactor-to-new-dependency-injection-features-in-symfony-3-3.md]: "current version" -> "current syntax" ?

**Comment by javiereguiluz** [comment]
👍  It looks correct to me. If you need more examples, check out the default app.yaml file defined by Symfony Flex: https://github.com/symfony/recipes/blob/master/symfony/framework-bundle/3.3/etc/packages/app.yaml

File: source/_posts/2017/2017-05-05-how-to-refactor-to-new-dependency-injection-features-in-symfony-3-3.md (hunk 5 above)
Comment (TomasVotruba) [inline source/_posts/2017/2017-05-05-how-to-refactor-to-new-dependency-injection-features-in-symfony-3-3.md]: I got some inner latte parsing issues (that are difficult to debug and not priority at the moment), so I put spaces there.

Will try without with sign hack then.

**Pull request merged**